include/kernels/madwf_transfer.cuh

```diff
@@ -0,0 +1,145 @@
+#pragma once
+
+#include <color_spinor_field.h>
+#include <color_spinor_field_order.h>
+#include <shared_memory_cache_helper.cuh>
+#include <madwf_transfer.h>
+#include <madwf_ml.h>
+
+/**
+  @file This file contains the argument and kernel that applies fifth dimension transfer to a vector:
+      out_s = T_{st} * in_t,
+  where s and t are fifth dimension indices. T_{st} is a spin matrix so T has shape Ls*4-by-Ls*4.
+*/
+
+namespace quda
+{
+
+  namespace madwf_ml
+  {
+
+    template <class real, int nSpin, int nColor, transfer_5D_t transfer_t> struct transfer_5D_mapper {
+    };
+
+    template <class real, int nSpin, int nColor> struct transfer_5D_mapper<real, nSpin, nColor, transfer_5D_t::Spin> {
+      using type = Matrix<complex<real>, nSpin>;
+    };
+
+    template <bool dagger, class real, int nSpin, int nColor>
+    __device__ __host__ inline ColorSpinor<real, nColor, nSpin>
+    matrix_vector_multiply(const Matrix<complex<real>, nSpin> &m, const ColorSpinor<real, nColor, nSpin> &v)
+    {
+      ColorSpinor<real, nColor, nSpin> out; // out is initialized to zero
+#pragma unroll
+      for (int color = 0; color < nColor; color++) {
+#pragma unroll
+        for (int column = 0; column < nSpin; column++) {
+          auto v_col = v(column, color);
+#pragma unroll
+          for (int row = 0; row < nSpin; row++) {
+            if (dagger) {
+              out(row, color) = cmac(conj(m(column, row)), v_col, out(row, color));
```

Comment:
Need to use `cmac` to generate optimal code.

Reply:
I must be drunk or something - these are already written `cmac` it seems. @maddyscientist

Reply:
I think I must be drunk too. I could have sworn these weren’t `cmac` when looking earlier 🤣